twine: add page (#8624)
tldr-pages/tldr: pages/common/twine.md

# twine

> > Twine is a utility for publishing Python packages on PyPI.

Reviewer: Just describe the command
```suggestion
> Utility for publishing Python packages on PyPI.
```

> > Subcommands such as `twine upload` have their own usage documentation.

Reviewer 2: What is the difference between the current command and `twine-upload`? Is this line necessary?

Author: I used `pages/common/pip.md` as a template which had a line similar to this one.

`twine` has two other subcommands `check` and `register`, which I believe are seldom used. But perhaps I should include one example of each?

Reviewer 2: > I used `pages/common/pip.md` as a template which had a line similar to this one.
> 
> `twine` has two other subcommands `check` and `register`, which I believe are seldom used. But perhaps I should include one example of each?

Thanks for the information, I will mark this conversation as resolved. As this line is applicable here.

Reviewer 3: There does not seem to be a `tldr` page `twine upload` so it's weird to refer to it as if it existed.

I would get rid of this sentence here, and if applicable/useful, add one point for usage of `check` and `register` each.

If this was ever broken out, we could then add `Subcommands such as `twine upload` have their own usage documentation.` but right now this will just confuse users, as they won't find a `tldr twine upload` page.

> More information: <https://twine.readthedocs.io>.

> - Upload to Test PyPI and verify things look right:

Author: The 'r' stands for 'repository'. So in the edit I changed this line to be:
```
- Upload to the Test PyPI [r]epository and verify things look right:
```

`twine upload -r testpypi dist/*`

> - Upload to PyPI:

Reviewer 3: I would put the Upload to PyPI first, before test upload.

The upload itself is after all the main usage, the test upload is optional.

`twine upload dist/*`

> - Specify username and password:

Author: Applied this locally, thanks!

> - Specify [u]sername and [p]assword while uploading to PyPI:

Reviewer: ```suggestion
- Upload to PyPI with a specified [u]sername and [p]assword:
```

`twine upload -u {{username}} -p {{password}} dist/*`

> - Upload to an alternative repository url:

Reviewer: alternative like a mirror? or another registry?
```suggestion
- Upload to an alternative repository URL:
```

Author: Could be any valid repository (package index) URL. For some cases, you might only upload to a single url:
```
twine upload --repository-url <my pypi url> dist/*
```

Or perhaps issue twine upload multiple times to upload to different urls
```
twine upload --repository-url <my first pypi url> dist/*
twine upload --repository-url <my second pypi url> dist/*
```
